test/models/host_test.rb

```diff
@@ -266,6 +266,23 @@ def teardown
       host.expects(:enable_orchestration!).never
       assert host.import_facts(raw['facts'])
     end
+
+    test 'should import facts boot time to report data facet' do
+      refute Host.find_by_name('sinn1636.lan')
+      raw = read_json_fixture('facts/facts_with_certname.json')
+      host = Host.import_host(raw['name'], 'puppet')
+      assert host.import_facts(raw['facts'])
+      first_boot_time = host.reported_data.boot_time
+      refute_nil host.reported_data.boot_time
+      assert_kind_of DateTime, host.reported_data.boot_time
```

Comment:
This fails, but I don't like asserting exact type match in dynamicaly typed language anyway.
Cant we predict the value and assert the values are equal? The equality should be handled more predictably.
Time is bit tricky, but something like this would maybe do?
```suggestion
      assert (Time.now - host.reported_data.boot_time - raw['facts']['uptime_seconds'].to_i.seconds) < 2.seconds
```

Reply:
I tried to do something smarter with freeze_time